File: day01/day1.1.rb
#! /usr/bin/env ruby -an

BEGIN {$sum=0}
END {puts "Calibration value: #{$sum}"}

$first = $_.match(/[^0-9]*([0-9]).*$/)[1]
$last  = $_.match(/.*([0-9])[^0-9]*$/)[1]

$sum += ($first+$last).to_i

File: day01/day1.1.swift
#!/usr/bin/env swift -enable-bare-slash-regex

let simpleDigits = /([0-9])/

var inputS: [String] = []
var sum = 0

while let input = readLine() {
    inputS.append(input)
}

for l in inputS {
    var num = 0
    if let match = l.firstMatch(of: simpleDigits) {
        num = Int(match.1) ?? 0
        sum = sum + num * 10
    }
    if let match = String(l.reversed()).firstMatch(of: simpleDigits) {
        num = Int(match.1) ?? 0
        sum = sum + num
    }
}
print("Total: \(sum).")
